src/cpptests/t_forkgc.cpp

```diff
@@ -14,6 +14,32 @@ static timespec getTimespecCb(void *) {
   return ts;
 }
 
+typedef struct {
+  RedisModuleCtx *ctx;
+  void *fgc;
+  IndexSpec *sp;
+} args_t;
+
+static bool runLoop;
+pthread_t thread;
```

Comment:
static!

Reply:
you actually don't need this, pthread_t just contains the thread id. if you're detaching it, you can just use it locally in the function.

Reply:
But it's probably a good idea to only allow one such thread to run at a time anyway